addon/src/main/java/com/vaadin/flow/component/charts/Chart.java

```diff
@@ -563,4 +584,63 @@ public Registration addYAxesExtremesSetListener(
         return addListener(YAxesExtremesSetEvent.class, listener);
     }
 
+    private class DrilldownHandler implements Serializable {
+        private final Deque<Series> stack = new LinkedList<>();
+
+        private void onDrilldown(DrilldownEvent details) {
+            if (getDrilldownCallback() == null) {
+                return;
+            }
+            final int seriesIndex = details.getSeriesItemIndex();
+            final int pointIndex = details.getItemIndex();
+            final Series series = resolveSeriesFor(seriesIndex);
+            DataSeriesItem item = null;
+            if (series instanceof DataSeries) {
+                DataSeries dataSeries = (DataSeries) series;
+                item = dataSeries.get(pointIndex);
+            }
+            final DrilldownDetails chartDrilldownEvent = new DrilldownDetails(
+                Chart.this, series, item, pointIndex);
+
+            Series drilldownSeries = getDrilldownCallback()
+                .handleDrilldown(chartDrilldownEvent);
+            if (drilldownSeries != null) {
+                stack.push(drilldownSeries);
+                callClientSideAddSeriesAsDrilldown(seriesIndex, pointIndex,
+                    drilldownSeries);
+            }
+        }
+
+        private void onDrillup(ChartDrillupEvent e) {
+            stack.pop();
+        }
+
+        private void callClientSideAddSeriesAsDrilldown(int seriesIndex,
+            int pointIndex, Series drilldownSeries) {
+            final String JS = "this.__callChartFunction($0, this.configuration.series[$1].data[$2], $3)";
+            wrapJSExpressionInTryCatchWrapper(JS, "addSeriesAsDrilldown", seriesIndex, pointIndex,
+                toJsonValue((AbstractConfigurationObject) drilldownSeries));
+        }
+
+        private void wrapJSExpressionInTryCatchWrapper(String expression, Serializable... args) {
```

Comment:
This looks useful for other JS executions, can we add it as static helper that just wraps the expression but doesn't execute it?